core/rpc/core/src/impl/build_tx.rs

```diff
@@ -1078,16 +1037,14 @@ impl<C: CkbRpc> MercuryRpcImpl<C> {
     ) -> InnerResult<TransactionCompletionResponse>
     where
         F: Fn(&'a MercuryRpcImpl<C>, Context, T, u64) -> Fut + Copy,
-        Fut: std::future::Future<
-            Output = InnerResult<(TransactionView, Vec<SignatureAction>, usize)>,
-        >,
+        Fut: std::future::Future<Output = InnerResult<(TransactionView, Vec<ScriptGroup>, usize)>>,
         T: Clone,
     {
         let mut estimate_fee = INIT_ESTIMATE_FEE;
         let fee_rate = fee_rate.unwrap_or(DEFAULT_FEE_RATE);
 
         loop {
-            let (tx_view, signature_actions, change_cell_index) =
+            let (tx_view, script_groups, change_cell_index) =
                 prebuild(self, ctx.clone(), payload.clone(), estimate_fee).await?;
             let tx_size = calculate_tx_size(tx_view.clone());
```

Comment:
unrelated to this PR, I think fn `calculate_tx_size` may be changed to `calculate_tx_size(&tx_view)` to removes this unnecessary clone.